config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_notebooksenvironments.notebooks.cnrm.cloud.google.com.yaml

```diff
@@ -43,76 +42,52 @@ spec:
     name: v1alpha1
     schema:
       openAPIV3Schema:
+        description: NotebooksEnvironment is the Schema for the NotebooksEnvironment
+          API
         properties:
           apiVersion:
-            description: 'apiVersion defines the versioned schema of this representation
+            description: 'APIVersion defines the versioned schema of this representation
               of an object. Servers should convert recognized schemas to the latest
-              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
             type: string
           kind:
-            description: 'kind is a string value representing the REST resource this
+            description: 'Kind is a string value representing the REST resource this
               object represents. Servers may infer this from the endpoint the client
-              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
             type: string
           metadata:
             type: object
           spec:
+            description: NotebooksEnvironmentSpec defines the desired state of NotebooksEnvironment
             properties:
               containerImage:
                 description: Use a container image to start the notebook instance.
                 properties:
                   repository:
-                    description: |-
-                      The path to the container image repository.
-                      For example: gcr.io/{project_id}/{imageName}.
+                    description: 'Required. The path to the container image repository.
+                      For example: `gcr.io/{project_id}/{image_name}`'
                     type: string
                   tag:
                     description: The tag of the container image. If not specified,
                       this defaults to the latest tag.
                     type: string
-                required:
-                - repository
                 type: object
               description:
                 description: A brief description of this environment.
                 type: string
               displayName:
                 description: Display name of this environment for the UI.
                 type: string
-              locationRef:
```

Comment:
Any idea what this locationRef is? Could this infer something other than the location for the parent URL?

Reply:
I'm not sure where the `locationRef` came from. It was part of a TF-based alpha resource, so I wouldn't be too worried about this field. I checked the proto resource, and it includes a standard location as part of the parent.

https://github.com/googleapis/googleapis/blob/b4f2cb5295c6cd7b4f0366514508f70304c48a21/google/cloud/notebooks/v1/environment.proto#L33